File: api/Models/PlayTable/PlayTableModel.cs
using System;
using DoppelkopfApi.Entities;

namespace DoppelkopfApi.Models.PlayTable
{
    public class PlayTableModel
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public PlayStatus Status { get; set; }
        public int CurrentPlayerPosition { get; set; }

        public int RoundCardsGiversPosition { get; set; }

        public int RundCount { get; set; }

        public DateTime RunStarted { get; set; }

        public GamesVariants GameVariant { get; set; }

        public bool ShuffleRound { get; set; }

        public int PlayerPosition { get; set; }

        public int PlayerId { get; set; }

        public bool DiamondsAceAsMaster { get; set; }
        public bool WeddingWithFirstColorCast { get; set; }

        public string TableIcon { get; set; }

        public int AdditionalWeddingPlayerId { get; set; }
    }
}

File: client/src/doppelkopf/models/play-table.model.ts
export class PlayTable {
  id?: number;
  name: string;
  withNiner = false;
  status: PlayStatus;
  currentPlayerPosition: number;
  roundCardsGiversPosition: number;
  playerId?: number;
  diamondsAceAsMaster: boolean;
  weddingWithFirstColorCast: boolean;
  tableIcon?: string;
  additionalWeddingPlayerId?: number;
  gameVariant: GamesVariants;
}

export enum PlayStatus {
  None = 'None',
  Run = 'Run',
  Stop = 'Stop',
  Pause = 'Pause',
  End = 'End',
  SelectGameVarian = 'SelectGameVarian',
  WaitForNextRund = 'WaitForNextRund',
  WinnersTime = 'WinnersTime',
  RadyToStart = 'RadyToStart',
}

export enum GamesVariants {
  None = 'None',
  Normal = 'Normal',
  Wedding = 'Wedding',
  Solo = 'Solo',
  QueenSolo = 'QueenSolo',
  JackSolo = 'JackSolo',
  AceSolo = 'AceSolo',
  SuitSolo = 'SuitSolo',
  KingSolo = 'KingSolo',

  ColoRSoloClubs = 'ColoRSoloClubs',
  ColoRSoloDiamonds = 'ColoRSoloDiamonds',
  ColoRSoloHearts = 'ColoRSoloHearts',
  ColoRSoloSpades = 'ColoRSoloSpades',
}
